### PR TITLE
mixer_module: add clockwise motor numbering configuration

### PR DESCRIPTION
**Pull request by cmic0**
This PR adds support to vehicles / 4in1 ESCs that have clock-wise motor ordering:
4     1
3     2

To enable this configuration `MOT_ORDERING` needs to be set to 2. 

To be consistent with PX4 documentation, motor_test numbering has also been unified to 1..8 instead of 0..7.

**Test data / coverage**
Tested on a vehicle (pixhawk4) where motors were wired accordingly the clock-wise convention.

### CODE REVIEW

**Comment by dagar** [comment]
I don't see why motor ordering deserves special treatment outside of the existing mixers.

**Comment by cmic0** [comment]
@dagar I originally created a new geometry file with the clock-wise convention but I actually think that being able to change that via the parameter is more clean. 
By doing so you can still refer to the `4x` geometry in the .mix instead of a custom one

File: src/systemcmds/motor_test/motor_test.cpp [hunk 1]
```diff
@@ -88,7 +88,7 @@ Note: this can only be used for drivers which support the motor_test uorb topic
 
 	PRINT_MODULE_USAGE_NAME("motor_test", "command");
 	PRINT_MODULE_USAGE_COMMAND_DESCR("test", "Set motor(s) to a specific output value");
-	PRINT_MODULE_USAGE_PARAM_INT('m', -1, 0, 7, "Motor to test (0...7, all if not specified)", true);
+	PRINT_MODULE_USAGE_PARAM_INT('m', -1, 1, 8, "Motor to test (1...8, all if not specified)", true);
```
Comment (jlecoeur) [inline src/systemcmds/motor_test/motor_test.cpp]: Can we keep 0 indexing here?

**Comment by bkueng** [comment]
> @bkueng did you consider adding a geometry instead of introducing MOT_ORDERING? I am just curious to know if it was because you did not think about it, or because a param was more user-friendly.

Yes I was aware of that. I wanted to use the same airframe configs, so the param simplifies both devel and user setups. I'm not a fan of it either though.

**Comment by dagar** [comment]
> I actually think that being able to change that via the parameter is more clean.

Yes there's no question it's cleaner. The problem is if we circumvent the existing architecture for immediate convenience it might have small short term repercussions and more subtly it can potentially make it harder to ever properly address these larger issues (airframe and mixer configurability).

https://docs.px4.io/master/en/airframes/airframe_reference.html#quadrotor-x

Screen Shot 2019-10-25 at 12 02 55 PM

**Comment by LorenzMeier** [comment]
I think we're starting to dig ourselves into a rabbit hole with this. @cmic0 instead of supporting everything under the sun, can this be done as a specific vehicle-specific mixer / configuration that is part of one setup? That way it is clean for the one outlier.

File: src/systemcmds/motor_test/motor_test.cpp (hunk 1 above)
Comment (cmic0) [reply src/systemcmds/motor_test/motor_test.cpp]: @jlecoeur I think having this command with 1..8 range is more compliant with what is specified in the documentation/reference images ( Motor 1, Motor 2 and so on, https://docs.px4.io/master/en/airframes/airframe_reference.html#quadrotor-x). In this view things are more uniform, no?

**Comment by cmic0** [comment]
Agreed. Closing this one for now.

**Pull request closed without merging**